File: bench/data_transform/data_transform.cpp
#include <iostream>
#include <vector>

// 10000個の構造体を作成・変換
struct Item { int id; int value; int doubled; };

int main() {
    std::vector<Item> items;
    items.reserve(10000);

    for (int i = 0; i < 10000; i++) {
        items.push_back({i, i, i * 2});
    }

    std::cout << items.size() << std::endl;
    return 0;
}

File: bench/data_transform/DataTransform.java
import java.util.List;
import java.util.ArrayList;
import java.util.stream.IntStream;

// 10000個のレコードを作成・変換
record Item(int id, int value, int doubled) {}

public class DataTransform {
    public static void main(String[] args) {
        List<Item> items = IntStream.range(0, 10000)
            .mapToObj(i -> new Item(i, i, i * 2))
            .toList();
        System.out.println(items.size());
    }
}
